`make -j` fails in `clash-vexriscv`

```
martijn@deskei:~/code/clash-vexriscv/clash-vexriscv$ make -j
cd example-cpu; sbt "runMain example.ExampleCpu"
g++ -I/nix/store/ba71fjpwqk0fh30fhi3k6iaagh2qaa2d-verilator-5.018/share/verilator/include -I/nix/store/ba71fjpwqk0fh30fhi3k6iaagh2qaa2d-verilator-5.018/share/verilator/include/vltstd -fPIC -O3 -Ibuild_out_dir/verilator_output -c src/ffi/impl.cpp -o build_out_dir/impl.o
g++ -I/nix/store/ba71fjpwqk0fh30fhi3k6iaagh2qaa2d-verilator-5.018/share/verilator/include -I/nix/store/ba71fjpwqk0fh30fhi3k6iaagh2qaa2d-verilator-5.018/share/verilator/include/vltstd -fPIC -O3 -Ibuild_out_dir/verilator_output -c /nix/store/ba71fjpwqk0fh30fhi3k6iaagh2qaa2d-verilator-5.018/share/verilator/include/verilated.cpp -o build_out_dir/verilated.o
src/ffi/impl.cpp:5:10: fatal error: VVexRiscv.h: No such file or directory
    5 | #include "VVexRiscv.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:39: build_out_dir/impl.o] Error 1
make: *** Waiting for unfinished jobs....
^Cmake: *** [Makefile:46: build_out_dir/verilated.o] Interrupt
make: *** [Makefile:23: example-cpu/VexRiscv.v] Error 130
```

--- a/clash-vexriscv/Makefile
+++ b/clash-vexriscv/Makefile
@@ -9,6 +9,8 @@ FFI_DIR = src/ffi
 
 VERILATOR_FLAGS = -CFLAGS '-O3 -fPIC' -Wno-fatal +1364-2001ext+v --trace
 
+N := $(patsubst -j%,%,$(filter -j%,$(MAKEFLAGS)))
+
 VERILATOR_CFLAGS = $(shell pkg-config --cflags verilator)
 FFI_CPPFLAGS = $(VERILATOR_CFLAGS) -fPIC -O3 -I$(VERILATOR_DIR)
 
@@ -31,19 +33,22 @@ $(VERILATOR_DIR)/VVexRiscv.mk $(VERILATOR_DIR)/VVexRiscv.h: $(OUT_DIR)/VexRiscv.
 	cd $(OUT_DIR); verilator $(VERILATOR_FLAGS) --cc -Mdir ../$(VERILATOR_DIR) VexRiscv.v
 
 $(VERILATOR_DIR)/VVexRiscv__ALL.a: $(VERILATOR_DIR)/VVexRiscv.mk
-	cd $(VERILATOR_DIR); make -f VVexRiscv.mk
+	cd $(VERILATOR_DIR); make -f VVexRiscv.mk -j $(N)
 
-$(OUT_DIR)/impl.o: $(FFI_DIR)/impl.cpp $(FFI_DIR)/interface.h
+$(OUT_DIR)/impl.o: $(FFI_DIR)/impl.cpp $(FFI_DIR)/interface.h $(VERILATOR_DIR)/VVexRiscv.h
 	$(CXX) $(FFI_CPPFLAGS) -c $(FFI_DIR)/impl.cpp -o $(OUT_DIR)/impl.o
 
 $(OUT_DIR)/verilated_vcd_c.o: $(shell pkg-config --variable=includedir verilator)/verilated_vcd_c.cpp
+	mkdir -p $(OUT_DIR)
 	$(CXX) $(FFI_CPPFLAGS) -c $(shell pkg-config --variable=includedir verilator)/verilated_vcd_c.cpp -o $(OUT_DIR)/verilated_vcd_c.o
 
 
 $(OUT_DIR)/verilated.o: $(shell pkg-config --variable=includedir verilator)/verilated.cpp
+	mkdir -p $(OUT_DIR)
 	$(CXX) $(FFI_CPPFLAGS) -c $(shell pkg-config --variable=includedir verilator)/verilated.cpp -o $(OUT_DIR)/verilated.o
 
 $(OUT_DIR)/verilated_threads.o: $(shell pkg-config --variable=includedir verilator)/verilated_threads.cpp
+	mkdir -p $(OUT_DIR)
 	$(CXX) $(FFI_CPPFLAGS) -c $(shell pkg-config --variable=includedir verilator)/verilated_threads.cpp -o $(OUT_DIR)/verilated_threads.o
 
 $(OUT_DIR)/VVexRiscv__ALL.a: $(VERILATOR_DIR)/VVexRiscv__ALL.a

--- a/clash-vexriscv/Setup.hs
+++ b/clash-vexriscv/Setup.hs
@@ -24,7 +24,7 @@ makeExtLib _ flags = do
   rawSystemExit
     verbosity
     "env"
-    ["make"]
+    ["make", "-j"]
   return emptyHookedBuildInfo
 
 updateExtraLibDirs :: LocalBuildInfo -> IO LocalBuildInfo

